src/pages/Promotions/Details/index.tsx

```diff
@@ -0,0 +1,229 @@
+        }
+      });
+    }
+  };
+
+  const initialValues: PromotionFormValue = {
+    name: data?.promotion?.name || "",
+    description: data?.promotion?.description || "",
+    promotionType: (data?.promotion?.promotionType || "order-discount") as PromotionType,
+    actions: data?.promotion?.actions || [],
+    triggers: data?.promotion?.triggers ? formatTriggers(
+      data.promotion.triggers,
+      data?.promotion?.name || "trigger name"
+    ) : [],
+    stackability: data?.promotion?.stackability || { key: "none", parameters: {} },
+    label: data?.promotion?.label || "",
+    startDate: data?.promotion?.startDate || null,
+    endDate: data?.promotion?.endDate || null
+  };
+
+  const showActionButtons = promotionId ? canUpdate : canCreate;
+
+  return (
+    isLoading ? <Loader/> :
+      <Formik<PromotionFormValue>
+        onSubmit={onSubmit}
+        initialValues={initialValues}
+        validationSchema={promotionSchema}
+      >
+        {({ values, dirty, resetForm, isSubmitting, submitForm }) => <Stack component={Form} direction="column" gap={3} pb={3}>
+          <Paper variant="outlined" square sx={{ padding: 3, pb: 0, position: "sticky", top: { xs: 56, sm: 64 }, zIndex: 1 }}>
+            <Stack direction="row" alignItems="center" justifyContent="space-between" pb={1}>
+              <Stack direction="column" maxWidth="80%" flexWrap="wrap">
+                <Typography variant="h6" gutterBottom>{values.name || "New Promotion"}</Typography>
+                <Stack direction="row" gap={0.5} alignItems="center">
+                  {data?.promotion?.referenceId ?
+                    <>
+                      <Typography
+                        variant="subtitle2"
+                        sx={{ color: "grey.600" }}>
+                        {data?.promotion.referenceId}
+                      </Typography>
+                      <FiberManualRecordIcon sx={{ height: "5px", width: "5px", color: "grey.600" }} />
+                    </>
+                    : null}
+
+                  <Typography
+                    variant="subtitle2"
+                    sx={{ color: "grey.600" }}
+                    noWrap
+                  >
+                    {PROMOTION_TYPE_OPTIONS[values.promotionType as PromotionType]?.label || "Unknown"}
+                  </Typography>
+                </Stack>
+              </Stack>
+              {showActionButtons ?
+                <ActionButtons
+                  loading={isSubmitting}
+                  disabled={!dirty}
+                  submitForm={submitForm}
+                  onCancel={() => (promotionId ? resetForm() : navigate("/promotions"))}
+                  promotionId={promotionId}
+                />
+                : null}
+            </Stack>
+            <Tabs value="details">
+              <Tab disableRipple value="details" label="Details"/>
+            </Tabs>
+          </Paper>
+          <PromotionSection title="Promotion Basics">
```

Comment:
The UI responsible for a promotion section seems like it should be abstracted away even further. It's a really common case across the app to have a card with a header. Why don't we use the component from `@components/Card` here? I vision it to be something like:
```typescript
import { PropsWithChildren } from "react";
import { CardContent, CardHeader, Card as MuiCard, Divider } from "@mui/material";

export type CardProps = {
  title: string
  action?: JSX.Element
  dense?: boolean
  divider?: boolean
}

export const Card = ({ title, action, dense = false, divider = false, children }: PropsWithChildren<CardProps>) => (
  <MuiCard variant="outlined">
    <CardHeader title={title} action={action} sx={{
      ...({ dense } && { py: 1 })
    }}/>
    {divider && <Divider/>}
    <CardContent>
      {children}
    </CardContent>
  </MuiCard>
);
```

We should have unified styling for the cards with headers across the app.

To match the already established styling to the header we can also edit the theme like so:
```typescript
    MuiCardHeader: {
      defaultProps: {
        titleTypographyProps: {
          variant: "subtitle1"
        }
      }
    }
```

Reply:
Sound like a good idea to reuse the Card component. I will take a look